### DISCUSSION TITLE
Are queries cached?

### DISCUSSION THREAD

**Discussion by nikoladev** [gatsbyjs/gatsby#27816] [Help] [answered]
Are the results of a page query or static query cached? For example, if I have a component that calls `useStaticQuery` and I use that component a lot, will the query be done every time the component is called or will the result of the query be cached?

I'm basically wondering if I need to add my own caching on top.

Thanks!

**Comment by nikoladev** [comment]
A related question: if I have a static query which I use in a lot of places, should I only call it in one place and then use React context to pass the result around? This is not for performance reasons, but more for DRY and maintainability reasons.

Or should I ask this in a separate thread?

**Comment by herecydev** [comment, marked as answer]
The entire query is hashed, if gatsby has already retrieved that hash it won't need to refetch it, so it is cached browser-side. You could either use react's `Context` or create a custom hook that wraps Gatsby's `useStaticQuery` and returns the result of your static query.

I would start with a custom hook, that's worked very well for me.

**Reply by nikoladev** [reply]
Thanks! This is exactly what I needed to know.

Great tip on the custom hook :)

**Reply by joernroeder** [reply]
I'm running into an issue where I have a component which contains a static query on a lot of pages. I'm wondering how that query gets cached during builds because my build time almost 10xed on gatsby cloud.

**Comment by arthuratmeetbreeze** [comment]
@herecydev do you know if there is a way to make gatsby refetch the page query including runing all custom resolvers created with schema customization during development. I'd like this to happen when I'm making changes to source files (pages, components, content sources, etc.) 

I have a case where I've got custom fields added to MarkdownRemark via `createSchemaCustomization` which fetches data outside of gatsby. My gatsby page query is not changed during development, but data that's being fetched and processed inside the field resolvers does change. 

So to make this work I eaither need to restart the development server or make and actual change in the page query (add, remove a random field) just to make it re-hash the page query and re-fetch the data by re-running the field resolvers.

Ideally it would be great if it also worked with hot-reloading, but I'd settle for a full refresh of the page just to make the page query refetch.

Ahy help would be appreciated greatly!!
